### ISSUE TITLE
Radio checkbox is not firing uncheck event

### ISSUE THREAD

**Issue by MarKayS** [Semantic-Org/Semantic-UI#4407]
I am using radiobutton and each choice will add some different new fields to the form. Therefore I need to know what exactly has change (what was the previous radio button and what is the new/current). I can get new radio button by listening for onChange or onCheck callback from checkbox(). However onChange/onUnchecked won't show that some radio has been automatically unchecked.

I found a quick workaround with using beforeChecked and finding current checked radiobutton, but I would like to know if this is intentional behaviour as I was suprised and found this counter-intuitive. It could be just me tho.

Fiddle if needed http://jsfiddle.net/v6qmf3fo/3/

PS: I love Semantic UI, keep up the great work !

**Comment by ilanus** [comment]
See: http://semantic-ui.com/modules/checkbox.html#/examples Everything works fine, I don't understand your issue

**Comment by MarKayS** [comment]
I should've comment out my workaround...

Let's work with this
http://jsfiddle.net/v6qmf3fo/6/http://jsfiddle.net/v6qmf3fo/6/

You see that in the info div only alternative unchecked fired (workaround), not the one stating unchecked only.

When I select some other radio button I would presume that onUnchecked would fire even tho I did not uncheck the radio itself, but it was unchecked automatically.

I just found this counterintuitive, but like I said, it could be just me, because I am used to this kind of behaviour from other frameworks. Doesn't really matter, because the workaround was quite simple.

**Comment by ilanus** [comment]
@MarKayS Yeah I can see what you mean, if you remove 'checked' property it will append 'undefined ' guess the loop starts at 1 somewhere :smile:

**Comment by MarKayS** [comment]
Yeah I am missing one if statement there. Point is, to my opinion, the "onUnchecked" should trigger, don't you think?

**Comment by ilanus** [comment]
That's a good question, @jlukic what do you think?

**Comment by dmsps** [comment]
Same issue, radio buttons dont uncheck if i check another.
